components/director/internal/repo/testdb/generic_repo_get_tests.go

```diff
@@ -15,16 +15,20 @@ import (
 // RepoGetTestSuite represents a generic test suite for repository Get method of any global entity or entity that has externally managed tenants in m2m table/view.
 // This test suite is not suitable entities with embedded tenant in them.
 type RepoGetTestSuite struct {
-	Name                      string
-	SQLQueryDetails           []SQLQueryDetails
-	ConverterMockProvider     func() Mock
-	RepoConstructorFunc       interface{}
-	ExpectedModelEntity       interface{}
-	ExpectedDBEntity          interface{}
-	MethodArgs                []interface{}
-	AdditionalConverterArgs   []interface{}
-	DisableConverterErrorTest bool
-	MethodName                string
+	Name                                  string
+	SQLQueryDetails                       []SQLQueryDetails
+	ConverterMockProvider                 func() Mock
+	RepoConstructorFunc                   interface{}
+	ExpectedModelEntity                   interface{}
+	ExpectedDBEntity                      interface{}
+	MethodArgs                            []interface{}
+	AdditionalConverterArgs               []interface{}
+	DisableConverterErrorTest             bool
+	MethodName                            string
+	ExpectNotFoundError                   bool
+	AfterNotFoundErrorSQLQueryDetails     []SQLQueryDetails
+	AfterNotFoundErrorExpectedModelEntity interface{}
+	AfterNotFoundErrorExpectedDBEntity    interface{}
```

Comment:
Doesn't seems to be very generic feature for the generic suite. The assumption that if "NotFoundError" is expected then a new query will be executed is hard to imagine to be reusable. Even if it is, it would be in so few places that the cognitive overhead to grasp those new long and specific arguments won't be worth it every-time the suite is used (and most of the time not needed).

It could be introduced a bit less static as now it is specific for just one test and pollutes the suite.
You can introduce only the `ExpectNotFoundError bool` and configure it, if it is true, then the `returns not found error when no rows` will be branched to not expect not found error.

Then in the `SQLQueryDetails` you can add one more query for the NotFoundCase and rely on the generic success and error cases that are generated to handle that for you.

If that does not work out for whatever reason (since I am not much into the details, but it seems doable), it is better to me to move the tests as additional tests in the specific repo. Otherwise with a few more such overly specific arguments the suites may become too complex to be used and their purpose would be defeated.